### PR TITLE
Add addWatchTarget() option to config

### PR DESCRIPTION
**Pull request by MadeByMike**
I'm trying to move away from build processes tacked onto the front of SSGs. This little helper makes this much easier. It allows you to add a custom watch target so that I can tell eleventy, when my scss or javascript files change, I want you to rebuild the site.

Here's how I use it:

```javascript
// Additional watch targets
config.addWatchTarget('./src/scss/'); 
config.addWatchTarget('./src/js/');
```

### CODE REVIEW

**Comment by MadeByMike** [comment]
The build passed but Travis wigged out... can we re-trigger it?

**Comment by zachleat** [comment]
Restarted!

**Comment by MadeByMike** [comment]
Don't get it... same thing happened. Windows test all pass the CI doesn't close and will timeout. I ran this locally on Windows and "Works on my Machine" - ideas?

**Comment by MadeByMike** [comment]
Yay \o/ looks like this was a know issue with Windows builds on Travis. Adding an env var `YARN_GPG=no` fixed it.

More discussion on this here: https://travis-ci.community/t/build-doesnt-finish-after-completing-tests/288

**Comment by MadeByMike** [comment]
@zachleat would it be too much to ping you again now that the build is passing?

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "workbench.colorCustomizations": {}
+}
```
Comment (zachleat) [inline .vscode/settings.json]: Can we remove this file?

**Comment by zachleat** [comment]
Before I merge this just one question: are the watch targets here relative to the input directory or the project root?

**Comment by MadeByMike** [comment]
@zachleat it eventually uses the `addRaw()` method of `EleventyWatchTargets.js` which I think means it's relative to the current working directory? Which is usually the same as the project root. Do you think we want to explicitly make this relative to something?

**Comment by ardouglass** [comment]
It looks like this solves #333 and #651. Can't wait for a new release. =)

**Comment by MadeByMike** [comment]
Yay @zachleat "Only those with write access to this repository can merge pull requests."

**Comment by zachleat** [comment]
Sorry Mike I’ve been swamped here I’ll try to get to this in the next few days!

**Comment by MadeByMike** [comment]
@zachleat <3 I know buddy. Take as long as you need.

**Comment by zachleat** [comment]
START THE PARADE

**Comment by zachleat** [comment]
Thank you @MadeByMike, this is a lovely addition 👍

**Comment by kevinmpowell** [comment]
I'm having trouble getting this to work with watching a single file:

```
  eleventyConfig.addWatchTarget("./components");
```
Works, but I don't want to watch that entire directory

```
  eleventyConfig.addWatchTarget("./components/concatenated.njk");
```
☝️ does not work. Can anyone else duplicate?

**Comment by zachleat** [comment]
Hey @kevinmpowell please file a new issue!

**Pull request merged**